pkgs/development/libraries/xxHash/default.nix

```diff
@@ -11,21 +17,25 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-2WoYCO6QRHWrbGP2mK04/sLNTyQLOuL3urVktilAwMA=";
   };
 
-  # Upstream Makefile does not anticipate that user may not want to
-  # build .so library.
-  postPatch = lib.optionalString stdenv.hostPlatform.isStatic ''
-    sed -i 's/lib: libxxhash.a libxxhash/lib: libxxhash.a/' Makefile
-    sed -i '/LIBXXH) $(DESTDIR/ d' Makefile
-  '';
+  # Merged in https://github.com/Cyan4973/xxHash/pull/649
+  # Should be present in next release
+  patches = [(fetchurl {
+    name = "cmakeinstallfix.patch";
+    url = "https://github.com/Cyan4973/xxHash/commit/636f966ecc713c84ddd3b7ccfde2bfb2cc7492a0.patch";
+    hash = "sha256-fj+5V5mDhFgWGvrG1E4fEekL4eh7as0ouVvY4wnIHjs=";
+  })];
 
-  outputs = [ "out" "dev" ];
 
-  makeFlags = [ "PREFIX=$(dev)" "EXEC_PREFIX=$(out)" ];
+  nativeBuildInputs = [
+    # Using unofficial CMake build script to install CMake module files.
+    cmake
+  ];
 
-  # pkgs/build-support/setup-hooks/compress-man-pages.sh hook fails
-  # to compress symlinked manpages. Avoid compressing manpages until
-  # it's fixed.
-  dontGzipMan = true;
+  cmakeDir = "../cmake_unofficial";
+
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=${if buildSharedLibs then "ON" else "OFF"}"
```

Comment:
This can probably use `stdenv.hostPlatform.isStatic` as well.

Reply:
Don't we want to keep the option though? A lot of nix packages does seem to have this pattern (although `buildSharedLibs` should probably be renamed to `enableStatic`).